qtwebengine: Use system ffmpeg on aarch64

Thra11 opened the pull request:
On aarch64, linking against the vendored ffmpeg fails. Including ffmpeg
as a dependency and passing -system-ffmpeg to qmake fixes this.

###### Motivation for this change
QtWebengine fails to build on aarch64 due to problems linking against the bundled ffmpeg.

###### Things done
Added a qmake flag `-system-ffmpeg` when building for Aarch64 or Aarch32
Added ffmpeg as an optional dependency

Note: So far, I have only tested Qt 5.11's qtwebengine.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @matthewbauer

Thra11 committed:
qtwebengine: Use system ffmpeg on aarch64

On aarch64, linking against the vendored ffmpeg fails. Including ffmpeg
as a dependency and passing -system-ffmpeg to qmake fixes this.

Slightly odd conditional in qmakeFlags to avoid altering the list on
non-arm platforms, so that the change doesn't trigger an unneccessary
rebuild.

dezgeg commented:
Maybe it could be done on all platforms? Having the build time of qtwebengine go down would be pretty useful.

Thra11 commented:
Yes, I wondered about doing it on all platforms. I decided (at least initially) to err on the side of caution and not risk breaking things which currently work.

ttuegel commented on pkgs/development/libraries/qt-5/modules/qtwebengine.nix:
```diff
@@ -117,7 +118,9 @@ EOF
     fi
    '';
 
-  qmakeFlags = optional enableProprietaryCodecs "-- -proprietary-codecs";
```
This is causing a rebuild on other platforms (I think) because the number of list elements changes. Until we decide to use `-system-ffmpeg` on all platforms, could you change this so that it does not rebuild on other platforms?

Thra11 replied:
I've just noticed I also made a mistake in my addition to propagatedBuildInputs, which means ffmpeg gets added on other platforms, when it shouldn't.

Thra11 commented:
I've made a few changes:
* Rephrased the qmakeFlags expression so that it evaluates to exactly the same list as before on non arm platforms
* Fixed the change to propagatedBuildInputs (previously it said `(!stdenv.isAarch32 || stdenv.isAarch64)`)
* Replaced `stdenv.isAarch*` with stdenv.hostPlatform.isArm so as to be correct for cross-compilation

Thra11 commented:
Fixed mistake: Replaced `isArm`, as it is a synonym for `isAarch32` and so doesn't cover `isAarch64`.
